site/source/tutorials/1040_data_driven_widgets/index.md

```diff
@@ -0,0 +1,193 @@
+
+{% section %}
+
+## Connecting the List to data
+
+{% task 'Connect the `List` to data using its `properties`.' %}
+
+Traditional widget frameworks like Dojo 1 required developers to tightly couple widget instances to data store instances. For example, it was common for Dojo 1 widget code to expect a `store` property to exist on a widget instance, and to further expect that store to expose a Dojo-specific data store API. While effective, this method of explicitly connecting widgets to data stores is brittle and forces widgets to have knowledge of a store implementation.
+
+Data-driven widgets in Dojo 2 don't need to know any details about underlying data sources. Instead of calling methods on a store directly, widgets request data from their parent widget using callback properties, and the parent passes properties containing relevant data back down to children. **Dojo 2 widgets can use this parent-driven data approach to enable compatibility with virtually any provider by strictly decoupling widgets from the stores that power them.**
+
+The first step to connecting the `List` to worker data is to update its `properties` interface so it can accept a `data` array.
+
+{% instruction 'Update the `ListProperties` so that a `data` array property can be passed into the `List`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/List.ts' lines:6,8,9-13 %}
+
+{% aside 'Playing it safe' %}
+The `WorkerProperties` interface is imported from the `Worker` widget so that the TypeScript typings for the `data` array can be as strict as possible.
+{% endaside %}
+
+The `ListProperties` interface now defines an optional `data` property that can be passed into a `List` widget. This property could've been called by any name other than `data`; the important takeaway is that data items can now be passed into the `List` widget using `properties`. The next step is to update the `List` to render a list of items based on its new `data` property.
+
+{% instruction 'Try to update the `List` so it also renders a `<div>` for each item in its `data` array property.' %}
+
+{% solution showsolution3 %}
+```ts
+protected renderItems() {
+    const { data = [] } = this.properties;
+    return data.map((item: any) => v('div', [ `${item.firstName} ${item.lastName}` ]));
+}
+
+protected render() {
+    return v('div', [
+        w(TextInput, {
+            value: this.properties.value,
+            onInput: this.onInput,
+            placeholder: 'Filter workers...'
+        }),
+        v('div', this.renderItems())
+    ]);
+}
+```
+{% endsolution %}
+
+The `List` widget now renders both a `TextInput` that a user can type into and a list of result items based on its `data` property. Item rendering is offloaded into a helper method (`renderItems`) so widgets that extend `List` that need to modify how items are rendered only have to override a small helper method instead of the main `render` method.
```

Comment:
so widgets that extend `List` that need to modify how items are rendered only have to override a small helper method instead of the main `render` method -> so widgets extending `List` that need to modify the rendering of items only need to override a small helper method instead of the main `render` method